File: ALG3/lucky_code/Hanoi_disks/Stack.java
import java.util.List;
import java.util.LinkedList;

public class Stack<T> {

    private List<T> list;

    public Stack() {
        this.list = new LinkedList<>();
    }

    public void push(T value) {
        this.list.add(value);
    }

    public T pop() {
        if (this.list.isEmpty()) {
            System.out.println("error: stack is empty!");
            return null;
        }
        return this.list.remove(this.list.size() - 1);
    }

    public T get(int index) {
        if ((this.list.size() - 1) < index) {
            //System.out.println("error: stack is empty!");
            return null;
        } else {
            return this.list.get(index);
        }
    }

    public boolean isEmpty() {
        return this.list.isEmpty();
    }

    public T top() {
        if (this.list.isEmpty()) {
            System.out.println("error: stack is empty!");
            return null;
        }

        return this.list.get(this.list.size() - 1);
    }
    
    public static void main(String[] args) {
        System.out.println("blblbl :-)");
    }
}

File: DEV3/c-cpp/CPP_first/Ch. 7 - Conteneurs standards/codes/stack.cpp
#include <list>
#include <stack>
#include <iostream>

using namespace std;

int main()
{	
    stack<int> q;
	//stack<int, list<int> > q;

	for(int i = 0; i < 10; i++)
		q.push(i * i);

	while(! q.empty())
	{
		cout << q.top() << " ";
		q.pop();
	}
	cout << endl;
	
	for(int i = 0; i < 10; i++)
		q.push(i * i);
	q.top() = 99;
		
	for(int i = 0; i < q.size(); i++)//sneaky loop is sneaky
	{
		cout << q.top() << " ";
		q.pop();
	}
	cout << endl << q.size() << endl;
}
